File: scala/hamming/src/main/scala/hamming.scala
object Hamming {

  def compute(strand1: String, strand2: String): Integer = {
    require(strand1.length == strand2.length, "Strands must be equal lengths.")

    strand1.zip(strand2).
      count{case (n1, n2) => n1 != n2}
  }

}

File: ruby/hamming/hamming.rb
module Hamming

  def self.compute(strand1, strand2)
    raise ArgumentError, "strand lengths must be equal" if strand1.size != strand2.size

    combine_strands(strand1, strand2).count do |nucleotide1, nucleotide2|
      nucleotide1 != nucleotide2
    end
  end

  def self.combine_strands(strand1, strand2)
    strand1.chars.zip(strand2.chars)
  end

end

module BookKeeping
  VERSION = 3
end
